cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -404,7 +430,74 @@ private SqlSelect toSql(CQLSortNode node) throws QueryValidationException {
       order.append(wrapForLength(wrapInLowerUnaccent(vals.getIndexText(), modifiers))).append(desc).append(", ")
       .append(wrapInLowerUnaccent(vals.getIndexText(), modifiers)).append(desc);
     }
-    return new SqlSelect(where, order.toString());
+    if (column == null) {
+      return new SqlSelect(where, order.toString(), null);
+    }
+    String fullQuery = null;
+    if (node.getSortIndexes().size() == 1 && limit > 0 && offset >= 0) {
```

Comment:
Consider to invert the if expression and have the short code block of the current else clause at the beginning:
```
if (node.getSortIndexes().size() != 1 || limit <= 0 || offset < 0) {
  StringBuilder sql = ...
  return new SqlSelect(where, order.toString(), sql.toString());
}
ModifierSet modifierSet = ...
```
This will also reduce the indentation of the optimized path.